binaries/runtime/src/lib.rs

```diff
@@ -0,0 +1,187 @@
+
+use std::{
+    collections::{BTreeSet, HashMap},
+    mem,
+};
+use tokio::{runtime::Builder, sync::mpsc};
+use tokio_stream::{wrappers::ReceiverStream, StreamMap};
+
+mod operator;
+
+pub fn main() -> eyre::Result<()> {
+    set_up_tracing().context("failed to set up tracing subscriber")?;
+
+    let node_id: NodeId = {
+        let raw =
+            std::env::var("DORA_NODE_ID").wrap_err("env variable DORA_NODE_ID must be set")?;
+        serde_yaml::from_str(&raw).context("failed to deserialize operator config")?
+    };
+    let communication_config: CommunicationConfig = {
+        let raw = std::env::var("DORA_COMMUNICATION_CONFIG")
+            .wrap_err("env variable DORA_COMMUNICATION_CONFIG must be set")?;
+        serde_yaml::from_str(&raw).context("failed to deserialize communication config")?
+    };
+    let operators: Vec<OperatorDefinition> = {
+        let raw =
+            std::env::var("DORA_OPERATORS").wrap_err("env variable DORA_OPERATORS must be set")?;
+        serde_yaml::from_str(&raw).context("failed to deserialize operator config")?
+    };
+
+    let mut communication: Box<dyn CommunicationLayer> =
+        communication::init(&communication_config)?;
+
+    let mut operator_events = StreamMap::new();
+    let mut operator_stop_publishers = HashMap::new();
+    let mut operator_events_tx = HashMap::new();
+
+    for operator_config in &operators {
+        let (events_tx, events) = mpsc::channel(1);
+        let stop_publisher = publisher(
+            &node_id,
+            operator_config.id.clone(),
+            STOP_TOPIC.to_owned().into(),
+            communication.as_mut(),
+        )
+        .with_context(|| {
+            format!(
+                "failed to create stop publisher for operator {}",
+                operator_config.id
+            )
+        })?;
+        operator_stop_publishers.insert(operator_config.id.clone(), stop_publisher);
+
+        operator_events.insert(operator_config.id.clone(), ReceiverStream::new(events));
+        operator_events_tx.insert(operator_config.id.clone(), events_tx);
+    }
+
+    let operator_events = operator_events.map(|(id, event)| Event::Operator { id, event });
+    let node_id_clone = node_id.clone();
+    let stop_thread = std::thread::spawn(move || {
+        Builder::new_current_thread()
+            .enable_all()
+            .build()
+            .unwrap()
+            .block_on(run(
+                node_id_clone,
+                operator_events,
+                operator_stop_publishers,
+            ))
+            .unwrap();
+    });
+
+    for operator_config in &operators {
+        let events_tx = operator_events_tx.get(&operator_config.id).unwrap();
+        spawn_operator(
+            &node_id,
+            operator_config.clone(),
+            events_tx.clone(),
+            communication.as_mut(),
+        )
+        .wrap_err_with(|| format!("failed to init operator {}", operator_config.id))?;
```

Comment:
Because of  the changes to `operator/python.rs`, this `spawn_operator` call will block until the operator is finished for Python operators. So operators that should be launched after the first Python operator would be delayed until that operator is finished. Or am I missing something?

Reply:
Yes. Maybe if I join the python operator at the end. It could be possible to have multiple python operator on the same runtime.